libs/designsystem/src/lib/components/action-sheet/action-sheet.component.spec.ts

```diff
@@ -0,0 +1,129 @@
+
+describe('ActionSheetComponent', () => {
+  let spectator: SpectatorHost<ActionSheetComponent>;
+  let popout;
+  const openDelayInMs = ActionSheetComponent.OPEN_DELAY_IN_MS;
+  const modalControllerSpy = jasmine.createSpyObj('ModalController', ['showActionSheet']);
+  const mockPlatformService = { isTouch: () => false };
+  const createHost = createHostFactory({
+    component: ActionSheetComponent,
+    declarations: [
+      MockComponent(IconComponent),
+      MockComponent(ButtonComponent),
+      MockComponent(ActionSheetPopoutComponent),
+    ],
+    providers: [
+      {
+        provide: WindowRef,
+        useValue: window,
+      },
+      {
+        provide: ModalController,
+        useValue: modalControllerSpy,
+      },
+      {
+        provide: PlatformService,
+        useValue: mockPlatformService,
+      },
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createHost('<kirby-action-sheet></kirby-action-sheet>');
+
+    spectator.component.header = 'Test header';
+    spectator.component.subheader = 'Test subheader';
+    spectator.component.items = [
+      { id: '1', text: 'Action 1' },
+      { id: '2', text: 'Action 2' },
+      { id: '3', text: 'Action 3' },
+    ];
+    spectator.component.cancelButtonText = 'Test cancel button text';
+    popout = spectator.element.getElementsByTagName('kirby-action-sheet-popout')[0];
+  });
+
+  it('should create', () => {
+    expect(spectator.component).toBeTruthy();
+  });
+
+  describe('icon', () => {
+    it('should have default icon', () => {
+      expect(spectator.component.triggerIconName).toEqual('more');
+    });
+
+    it('should support custom icon', () => {
+      const iconName = 'pension';
+      spectator.setInput('triggerIconName', iconName);
+      const icon = spectator.query(IconComponent);
+      expect(icon.name).toEqual(iconName);
+    });
+  });
+
+  describe('text', () => {
+    it('should have no initial text', () => {
+      const button = spectator.query('button');
+      expect(button.textContent).toEqual('');
+    });
+
+    it('should render custom button text', () => {
+      const buttonText = 'Custom text';
+      spectator.setInput('triggerText', buttonText);
+      const button = spectator.query('button');
+      expect(button.textContent).toEqual(buttonText);
+    });
+  });
+
+  describe('popout', () => {
+    beforeEach(() => {
+      mockPlatformService.isTouch = () => false;
+    });
+
+    it('should open and focus on click', fakeAsync(() => {
+      spectator.click('button');
+      tick(openDelayInMs);
+      expect(spectator.component.isOpen).toBeTruthy();
+      expect(spectator.element).toBeFocused();
+    }));
+
+    it('should toggle closed when clicked and open', () => {
+      spectator.component['state'] = OpenState.open;
+      spectator.detectChanges();
+      expect(popout).toHaveComputedStyle({ display: 'block' });
+
+      spectator.click('button');
+      popout = spectator.query('kirby-action-sheet-popout');
```

Comment:
Why do you need to re-query?

Reply:
Who knows 🤔 😁 
<img width="901" alt="Screenshot 2021-04-16 at 14 06 22" src="https://user-images.githubusercontent.com/909794/115022131-3fb89100-9ebd-11eb-955d-b130304f35bd.png">

Reply:
Removed